### PR TITLE
add error check of i2c_write_read()

### PR DESCRIPTION
**Pull request by Hxinrong**
### Contribution description
drivers/ethernet/eth_sam_gmac.c  :  add error check of i2c_write_read() in function get_mac_addr_from_i2c_eeprom()

### Signed-off
Signed-off-by: Xinrong Han hanxr19@mails.tsinghua.edu.cn

### Issue
Fixes parts of #30205

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
LGTM, just fix the commit message to be proper, it is missing proper subject, body and signed-off-by

File: drivers/led/ht16k33.c [hunk 1]
```diff
@@ -218,7 +218,7 @@ uint32_t ht16k33_get_pending_int(const struct device *dev)
 	err = i2c_write_read(data->i2c, config->i2c_addr, &cmd, sizeof(cmd),
 			     &flag, sizeof(flag));
 	if (err) {
-		LOG_ERR("Failed to to read HT16K33 IRQ flag");
+		LOG_ERR("Failed to read HT16K33 IRQ flag");
```
Comment (tbursztyka) [inline drivers/led/ht16k33.c:221]: make that change in another commit, as it is unrelated to the other change

File: drivers/ethernet/eth_sam_gmac.c [hunk 2]
```diff
@@ -1785,16 +1785,20 @@ static void get_mac_addr_from_i2c_eeprom(uint8_t mac_addr[6])
...
 
 	dev = device_get_binding(CONFIG_ETH_SAM_GMAC_MAC_I2C_DEV_NAME);
 	if (!dev) {
 		LOG_ERR("I2C: Device not found");
 		return;
 	}
 
-	i2c_write_read(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
+	err = i2c_write_read(dev, CONFIG_ETH_SAM_GMAC_MAC_I2C_SLAVE_ADDRESS,
 		       &iaddr, CONFIG_ETH_SAM_GMAC_MAC_I2C_INT_ADDRESS_SIZE,
```
Comment (tbursztyka) [inline drivers/ethernet/eth_sam_gmac.c:1797]: re-indent the parameters, below last '(' + 1 space

**Review by tbursztyka — CHANGES_REQUESTED** [review]

**Comment by jukkar** [comment]
Note that you have to fix the actual commit, not the PR description. Please please do the changes in you local tree and then just force push the changes back to this PR.

Suggestion for the commit message:
```
drivers: ethernet: gmac: check errors for i2c_write_read()

Validate that i2c_write_read() succeeds.

Fixes #30205

Signed-off-by: Xinrong Han hanxr19@mails.tsinghua.edu.cn
```

**Review by jukkar — APPROVED** [review]

File: drivers/led/ht16k33.c (hunk 1 above)
Comment (Hxinrong) [reply drivers/led/ht16k33.c]: Mh... I mixed up the branches just now, and I have moved the typo fix commit to PR #30733.
